pkg/ccl/backupccl/schedule_exec.go

```diff
@@ -257,22 +258,63 @@ func (e *scheduledBackupExecutor) backupSucceeded(
 		return nil
 	}
 
-	s, err := jobs.LoadScheduledJob(ctx, env, args.UnpauseOnSuccess, ex, txn)
+	unpausedSchedule, err := jobs.LoadScheduledJob(ctx, env, args.UnpauseOnSuccess, ex, txn)
 	if err != nil {
 		return err
 	}
-	s.ClearScheduleStatus()
-	if s.HasRecurringSchedule() {
-		if err := s.ScheduleNextRun(); err != nil {
+	unpausedSchedule.ClearScheduleStatus()
+	if unpausedSchedule.HasRecurringSchedule() {
+		if err := unpausedSchedule.ScheduleNextRun(); err != nil {
 			return err
 		}
 	}
-	if err := s.Update(ctx, ex, txn); err != nil {
+	unpausedArgs := &ScheduledBackupExecutionArgs{}
+	if err := pbtypes.UnmarshalAny(unpausedSchedule.ExecutionArgs().Args, unpausedArgs); err != nil {
+		return errors.Wrap(err, "un-marshaling args")
+	}
+	if unpausedArgs.ChainProtectedTimestampRecords {
+		// We expect the full backup that is un-pausing the incremental schedule to
+		// have a pts record that it can pass on to the incremental schedule. If
+		// that is not the case pause the schedule and fail the backup job.
+		if !args.ChainProtectedTimestampRecords || args.ProtectedTimestampRecord == nil {
+			errMsg := fmt.Sprintf("schedule %d cannot chain protected timestamp records since"+
+				" prior schedule %d does not have a protected timestamp record", unpausedSchedule.ScheduleID(),
+				schedule.ScheduleID())
+			schedule.Pause()
+			schedule.SetScheduleStatus(errMsg)
+			e.metrics.NumFailed.Inc(1)
+			log.Errorf(ctx, "failed to chain protected timestamps: %q", errMsg)
+			return nil
+		}
+
+		// Full backup schedule stores its protected timestamp record in the
+		// incremental schedule so that the record can be used by the subsequent
+		// incremental backup job. It bootstraps the chaining of pts records which
+		// hereafter occurs on the incremental schedule.
+		ptsRecord := *args.ProtectedTimestampRecord
+		unpausedArgs.ProtectedTimestampRecord = &ptsRecord
+		any, err := pbtypes.MarshalAny(unpausedArgs)
+		if err != nil {
+			return errors.Wrap(err, "marshaling args")
+		}
+		unpausedSchedule.SetExecutionDetails(
+			unpausedSchedule.ExecutorType(),
+			jobspb.ExecutionArguments{Args: any},
+		)
+	}
+
+	if err := unpausedSchedule.Update(ctx, ex, txn); err != nil {
 		return err
 	}
 
 	// Clear UnpauseOnSuccess; caller updates schedule.
 	args.UnpauseOnSuccess = jobs.InvalidScheduleID
+
+	// Subsequent full backups do not need to concern themselves with chaining
+	// of pts records. The incremental schedule will continue the chaining of
+	// the records.
```

Comment:
Hmm, is this actually sufficient? Consider a schedule with full backups that take 2h to run that are run daily, and incrementals are run every 10 minutes. After we've done the first full backup, does this scheme hand over the responsibility for timestamp protection to the incrementals? Would we run into an issue where if full backups take 2h and incrementals run every 10min, the first incremental to see the new backup would need to backup all the changes for the past 2h but will only have the last 10 min protected?